### PR TITLE
Refresh option in CLI

### PR DESCRIPTION
**Pull request by thomas-gerber**
# Description

Refresh is serial at the moment

## Type of change
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# Checklist
(Delete what does not apply)
- [x] Have you checked to there aren't other open Pull Requests for the same update/change?
- [x] Have you lint your code locally before submission?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully run tests with your changes locally?

### CODE REVIEW

**Comment by thomas-gerber** [comment]
@ypc-faros if you know how to properly parallelize the refreshes (in a way that makes the waiting look correct, i.e. with 2 "progress bars"), let me know

**Comment by tovbinm** [comment]
@thomas-gerber https://www.npmjs.com/package/p-limit

**Comment by thomas-gerber** [comment]
> @thomas-gerber https://www.npmjs.com/package/p-limit

That doesn't help. I want to run the refresh in parallel, and track each with its own ProgressBar. Right now, if I force parallelism, all the progress bars write to the same (current) console line.

**Comment by thomas-gerber** [comment]
> > @thomas-gerber https://www.npmjs.com/package/p-limit
> 
> That doesn't help. I want to run the refresh in parallel, and track each with its own ProgressBar. Right now, if I force parallelism, all the progress bars write to the same (current) console line.

I found a setup that works reasonably well: there are N progress bars (as many as active sources), all mangled up; but since we clear the bars when their source are refreshed, it is invisible to the user.

File: cli/src/refresh/run.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import {Metabase} from '../metabase/metabase-client';
+import {display, Emoji} from '../utils';
+
+interface RefreshConfig {
+  readonly airbyte: Airbyte;
+  readonly metabase: Metabase;
+}
+
+export function makeRefreshCommand(): Command {
+  const cmd = new Command().name('refresh');
```
Comment (tovbinm) [inline cli/src/refresh/run.ts:17]: so how about `refresh-sources`?

**Review by tovbinm — APPROVED** [review]

File: cli/src/airbyte/airbyte-client.ts [hunk 2]
```diff
@@ -145,7 +117,61 @@ export class Airbyte {
     }
   }
 
+  private async track(
+    job: number,
+    connectionId: string,
+    sourceName: string
+  ): Promise<void> {
+    const syncBar = new ProgressBar(':bar', {
+      total: 2,
```
Comment (tovbinm) [inline cli/src/airbyte/airbyte-client.ts:126]: you can probably use p-limit with defined parallelism + set `total` to the same number here

**Review by ypc-faros — APPROVED** [review]
LGTM 🥳 

This looks great! Maybe we can give https://www.npmjs.com/package/multi-progress-bars a shot as a possible improvement afterwards

**Comment by tovbinm** [comment]
`multi-progress-bars` seems to be an unpopular package. I would avoid it.

File: cli/src/airbyte/airbyte-client.ts (hunk 2 above)
Comment (thomas-gerber) [reply cli/src/airbyte/airbyte-client.ts:126]: max sources to be refreshed is 4, and they can all be concurrent

**Pull request merged**